wasm-glow-ballers-js/src/lib.rs

```diff
@@ -1,8 +1,42 @@
 use wasm_bindgen::prelude::*;
-use wasm_glow_ballers_lib::mapping;
+use wasm_glow_ballers_lib::{
+    MappingItemType, mapping_exports_with_type, mapping_imports_with_type, mapping_with_type,
+};
 
 #[wasm_bindgen]
-pub fn analyse(target: Vec<u8>) -> Result<JsValue, JsError> {
-    let map = mapping(&target).map_err(JsError::from)?;
+pub enum JsMappingItemType {
```

Comment:
Preferably stable & exported types implement certain 'minimal' traits that are widely used by others. For this type, I can think of deriving the following:
`#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]`